routemaster/jobs/batch.rb

```diff
@@ -0,0 +1,57 @@
+require 'routemaster/jobs'
+require 'routemaster/models/batch'
+require 'routemaster/services/deliver'
+require 'routemaster/mixins/log'
+
+module Routemaster
+  module Jobs
+    class Batch
+      include Mixins::Log
+
+      def initialize(delivery: Services::Deliver)
+        @delivery = delivery
+      end
+
+      def call(uid)
+        batch = Models::Batch.new(uid: uid)
+
+        # prevents further ingestion in this batch (idempotent)
+        batch.promote
+
+        subscriber = batch.subscriber
+        data = batch.data
+
+        # handle unsubscription, autodrop
+        if subscriber.nil? || data.nil? || data.empty?
+          batch.delete
+          return self
+        end
+
+        events = data.
+          map { |d| Services::Codec.new.load(d) }.
+          select { |msg| msg.kind_of?(Models::Event) }
```

Comment:
Similarly here, I'd possibly make this an `events` method on `Batch` as it seems that _it_ should be what knows its own structure rather than jobs/services (to be honest, I'd be inclined to hide the internal data structure of a batch as that's an implementation detail and only expose the actual interface that's desired by clients (i.e. nobody outside `Models::Batch` should really be using the raw data).